### ISSUE TITLE
Emotion detection model

### ISSUE THREAD

**Issue by rlucatoor** [tanaos/artifex#22]
## What is an Emotion Detection model

An Emotion Detection model associates texts with labels describing the main emotion they convey.

## What needs to be done

Implement an Emotion Detection model that exposes the following public methods:

### `.train()` method:

```python
.train(output_path: Optional[str] = None, num_samples: int = config.DEFAULT_SYNTHEX_DATAPOINT_NUM, num_epochs: int = 3)`
```

The method should train the model to assign one of the following labels
- `joy`
- `anger`
- `fear`
- `sadness`
- `surprise`
- `disgust`
- `excitement`
- `neutral`

to any piece of text.

### `.__call__(text: Union[str, list[str])`

- If a `str` is provided, the model should return any one of the labels listed above.

- If a `list[str]` is provided, the model should return a list of labels, one per sentence.

### `.load(model_path: str)`

Loads a pre-trained Emotion Detection model from the specified path.
